freqtrade/rpc/rpc_manager.py

```diff
@@ -48,6 +49,17 @@ def __init__(self, freqtrade) -> None:
             apiserver.add_rpc_handler(self._rpc)
             self.registered_modules.append(apiserver)
 
+            # Enable External Signals mode
+            # For this to be enabled, the API server must also be enabled
+            if config.get('external_signal', {}).get('enabled', False):
+                logger.info('Enabling RPC.ExternalSignalController')
+                from freqtrade.rpc.external_signal import ExternalSignalController
+                external_signals = ExternalSignalController(self._rpc, config, apiserver)
+                self.registered_modules.append(external_signals)
+
+                # Attach the controller to FreqTrade
+                freqtrade.external_signal_controller = external_signals
```

Comment:
i REALLY don't like this from an architectural standpoint.

the bot should not have to care which rpc modules are attached - and i fear problems with teardown of the bot due to this - as it can lead to the refcount of objects to never reach 0 (it's technically a cyclic reference).

Reply:
This is one of the things I wanted your opinion on. This was done so the bot could have reference to the controller for running `emit_data`, but since I will be moving the bulk of the actual emits to the RPC module then it won't be needed.